netlogo-gui/src/main/lab/Worker.scala

```diff
@@ -107,8 +107,38 @@ class Worker(val protocol: LabProtocol)
         val world = ws.world
         var d = world.getDimensions
         for((name, value) <- settings) {
-          if(world.isDimensionVariable(name)) {
             val v = value.asInstanceOf[java.lang.Double].intValue
+            if(world.isDimensionVariable(name)) {
+            val pattern = "(min|max)-(pxcor|pycor)".r
+            val isValidDim = name match{
```

Comment:
I notice that you are only using `isValidDim._2` when `isValidDim._1` is `false`. Instead of having `isValidDim be a `(Boolean, String)`, why not have it be an `Option[String]`. A `None` indicates no error, a `Some(string)` indicates that the error is described by the string contained in the `Some`.